Sources/Classes/Helpers/Vendors/Vendor.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  Vendor.swift
+//  MetricsReporter
+//
+//  Created by Desu Sai Venkat on 23/11/23.
+//
+
+import Foundation
+#if os(iOS) || os(tvOS)
+import UIKit
+#elseif os(watchOS)
+import WatchKit
+#endif
+
+
+#if os(iOS) || os(tvOS)
+internal class PhoneVendor: Vendor {
+    override var osName: String {
+        return UIDevice.current.systemName
+    }
+}
+#endif
+
+#if os(macOS)
+internal class MacVendor: Vendor {
+    override var osName: String {
+        return "macOS"
+    }
+}
+#endif
+
+#if os(watchOS)
+internal class WatchVendor: Vendor {
+    override var osName: String {
+        return WKInterfaceDevice.current().systemName
+    }
+}
+#endif
+
+internal class Vendor {
+    var osName: String {
+        return "unknown"
+    }
+    var osVersion: String {
```

Comment:
I know this will be duplicated for all `Vendor` subclasses. Still, we should follow the convention. Take an empty string here and override the variable in every class. You can use the `Utils` class for generating the value.

Reply:
Addressed this